### ISSUE TITLE
Adaptive design with mat_discover

### ISSUE THREAD

**Issue by sgbaird** [sparks-baird/mat_discover#17]
## Feature request
`mat_discover` can be adapted into an adaptive design scheme by iteratively moving the top-ranked compound from the validation set to the training set and repeating the fit/predict. This is costly, as producing a top-10 list will now take 10x longer. While this could be mitigated by using algorithms that allow for "online" updating of the training data (probably out-of-scope for `mat_discover`), the extra computational cost is probably worth it for any kind of low-throughput simulation or experiment-based adaptive design scheme. It could also take the form of adding new simulation results to `mat_discover` and re-running.

The first idea would be an implementation within `mat_discover` with a docs example, while the latter would probably exist only as a docs example.

**Comment by sgbaird** [comment]
`Adapt` class implemented as of `2.0.0`
